x-pack/elastic-agent/pkg/agent/cmd/enroll.go

```diff
@@ -144,7 +148,25 @@ func enroll(streams *cli.IOStreams, cmd *cobra.Command, flags *globalFlags, args
 		return errors.New(err, "fail to enroll")
 	}
 
-	fmt.Fprintln(streams.Out, "Successfully enrolled the Agent.")
+	fmt.Fprintln(streams.Out, "Successfully enrolled the Elastic Agent.")
+
+	// skip restarting
+	noRestart, _ := cmd.Flags().GetBool("no-restart")
+	if noRestart {
+		return nil
+	}
+
+	daemon := client.New()
+	err = daemon.Start(context.Background())
+	if err == nil {
+		defer daemon.Stop()
```

Comment:
why are you stopping after restart is done?

Reply:
`Stop` just disconnects the client from the socket, it does not stop the running daemon.

Reply:
this all makes sense then :-)